src/data/boards.js

```diff
@@ -0,0 +1,190 @@
+const mongoCollections = require('../config/mongoCollections');
+const boards = mongoCollections.boards;
+const users = mongoCollections.users;
+const {colors} = require('../public/constants/index');
+const {ObjectId} = require('mongodb');
+const error_handler = require('../errors/error-handler'); 
+
+function newFields(newBoard, oldBoard) {
+  fields = {};
+  if(newBoard.boardName && newBoard.boardName !== oldBoard.boardName) {
+    if(!error_handler.checkNonEmptyString(newBoard.boardName))
+      throw new Error("Board name must not be empty.");
+    fields.boardName = newBoard.boardName;
+  }
+  if(newBoard.boardColor && newBoard.boardColor !== oldBoard.boardColor) {
+    if(!error_handler.checkColor(newBoard.boardColor))
+      throw new Error("Board color must be valid hex code.");
+    fields.boardColor = newBoard.boardColor;
+  }
+  if(newBoard.description && newBoard.description !== oldBoard.description) {
+    if(!error_handler.checkNonEmptyString(newBoard.description))
+      throw new Error("Board description must not be empty.");
+    fields.description = newBoard.description;
+  }
+
+  return fields;
+}
+
+
+module.exports = {
+  /**
+   * Get a board with a specific ID.
+   * @param {String} id The board's id.
+   * @returns A board.
+   */
+  readByID: async (id) => {
+    if(!error_handler.checkObjectId(id))
+      throw new Error("id is not valid.");
+
+    const boardCollection = await boards();
+    const board = await boardCollection.findOne({_id: ObjectId(id)});
+    if(board === null)
+      throw new Error("There is no board with that id.");
+
+    board['_id'] = board['_id'].toString();
```

Comment:
not familiar with [] notation here, I usually use dot notation for these objects. Not saying its wrong, just saying I can't validate with certainty.

Reply:
it's all G